applications/MedApplication/tests/test_med_model_part_io.py

```diff
@@ -4,38 +4,31 @@
 import KratosMultiphysics.MedApplication as KratosMed
 
 from KratosMultiphysics.kratos_utilities import DeleteFileIfExisting
+from testing_utilities import MedModelPartIOTestCase, GetMedPath, get_num_geometries_by_type
 
-from pathlib import Path
-
-
-
-def GetMedPath(med_path, med_name="mesh.med"):
-    return Path(__file__).absolute().parent / "med_files" / med_path / med_name
-
-
-class TestMedModelPartIO(KratosUnittest.TestCase):
 
+class TestMedModelPartIO(MedModelPartIOTestCase):
     def setUp(self):
         self.model = KM.Model()
         self.mp_read_1 = self.model.CreateModelPart("read_1")
         self.mp_read_2 = self.model.CreateModelPart("read_2")
 
-    def _execute_tests(self, med_path, check_fct):
+    def _execute_tests(self, med_path, check_fct, print_vtk=False):
         med_io_read_1 = KratosMed.MedModelPartIO(GetMedPath(med_path))
         med_io_read_1.ReadModelPart(self.mp_read_1)
 
-        self.assertGreaterEqual(KratosMed.MedTestingUtilities.ComputeLength(self.mp_read_1), 0.0)
-        self.assertGreaterEqual(KratosMed.MedTestingUtilities.ComputeArea(self.mp_read_1), 0.0)
-        self.assertGreaterEqual(KratosMed.MedTestingUtilities.ComputeVolume(self.mp_read_1), 0.0)
-        self.assertGreaterEqual(KratosMed.MedTestingUtilities.ComputeDomainSize(self.mp_read_1), 0.0)
+        self._basic_checks(self.mp_read_1)
+
+        if print_vtk:
+            write_vtk(self.mp_read_1, med_path)
 
         with self.subTest("check_model_part"):
             check_fct(self.mp_read_1)
 
         with self.subTest("read_write_read"):
             med_temp_path = GetMedPath(med_path, "temp.med")
-            DeleteFileIfExisting(med_temp_path) # make sure there are no leftovers from previous tests
-            self.addCleanup(DeleteFileIfExisting, med_temp_path) # clean up after test
+            DeleteFileIfExisting(med_temp_path)  # make sure there are no leftovers from previous tests
+            self.addCleanup(DeleteFileIfExisting, med_temp_path)  # clean up after test
```

Comment:
why not put it in `tearDown`? Idk, which one's better in this case, just asking why you chose `addCleanup`.

Reply:
if the test crashes mid-execution, then `tearDown` is not called. `addCleanup` on the other hand is still called, making sure that the file is removed. (Pretty sure it is like this, but its been years since I looked this up)